pkg/chunk/cache/memcached_client.go

```diff
@@ -30,27 +43,35 @@ type serverSelector interface {
 type memcachedClient struct {
 	*memcache.Client
 	serverList serverSelector
-	hostname   string
-	service    string
+
+	hostname string
+	service  string
+
+	addresses []string
+	provider  *dns.Provider
 
 	quit chan struct{}
 	wait sync.WaitGroup
+
+	numServers prometheus.Gauge
 }
 
 // MemcachedClientConfig defines how a MemcachedClient should be constructed.
 type MemcachedClientConfig struct {
-	Host           string        `yaml:"host"`
-	Service        string        `yaml:"service"`
-	Timeout        time.Duration `yaml:"timeout"`
-	MaxIdleConns   int           `yaml:"max_idle_conns"`
-	UpdateInterval time.Duration `yaml:"update_interval"`
-	ConsistentHash bool          `yaml:"consistent_hash"`
+	Host           string        `yaml:"host,omitempty"`
```

Comment:
We've just get rid of all `omitempty`. We shouldn't re-introduce them, or am I missing anything?